src/player.cpp

```diff
@@ -497,13 +503,14 @@ void player::process_turn()
     // Scent increases fast at first, and slows down as it approaches normal levels.
     // Estimate it will take about norm_scent * 2 turns to go from 0 - norm_scent / 2
     // Without smelly trait this is about 1.5 hrs. Slows down significantly after that.
+    if (scent < 0) {
+        scent = 0;
+    }
     if (scent < rng(0, norm_scent))
         scent++;
 
     // Unusually high scent decreases steadily until it reaches normal levels.
-    if (scent > norm_scent)
         scent--;
-
```

Comment:
Why the F I Have done this? It's been causing so much grief for me. I only have myself to blame...